File: Aula 08/Atividade.py
class SaldoInsuficienteError(Exception):
    """Erro utilizado quando o valor sacado é maior do que o que está amarzenado no saldo."""
    pass
    

class Conta_Base:
    def __init__(self):
        self.saldo = float(0)

class Conta_Corrente(Conta_Base):
    def __init__(self):
        super().__init__()

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteError(f"O valor sacado (R$ {valor:.2f}) é maior do que o saldo disponível (R$ {self.saldo:.2f}). Logo, o saque é inválido.")
        if valor <= self.saldo:
            self.saldo -= valor
            print(f"Saque completo. Saldo atual: {self.saldo:.2f}.")

conta = Conta_Corrente()
conta.saldo = 100

print(f"O saldo atual da conta é {conta.saldo:.2f}.")

try:
    conta.sacar(20)
except SaldoInsuficienteError as erro:
    print("Saldo incapaz de ser realizado: ", erro)
finally:
    print("Operação completa.")

File: Aula 08/Atividade.js
class SaldoInsuficienteError extends Error {
  constructor(message) {
    super(message);
    this.name = "SaldoInsuficienteError";
  }  
}

class contaBase {
    constructor(){
    this.saldo = 0
}
}

class contaCorrente extends contaBase {
    constructor() {
        super();
    }

    sacar(valor) {
        if (valor > this.saldo) {
            throw new SaldoInsuficienteError(
                `O valor sacado (R$ ${valor}) é maior do que o saldo disponível (R$ ${self.saldo}). Logo, o saque é inválido.`
            );
        };

        this.saldo -= valor;
        return `Saque completo. Saldo atual: ${self.saldo}.`
    };
};

const conta = new contaCorrente();
conta.saldo = 10;

console.log(`O saldo atual da conta é ${conta.saldo}`);

try {
    conta.sacar(20);
} catch(erro) {
    if (erro instanceof SaldoInsuficienteError) {
        console.log("Saldo incapaz de ser realizado: ", erro.message)
    };
    console.log(`Saque completo. Saldo atual: ${conta.saldo}.`);
    
} finally {
  console.log("Operação completa");
}